storage/StorageInterface.h

```diff
@@ -214,6 +214,23 @@ namespace storage
 
 
     /**
+     * Abstract class for some callbacks during commit. Currently only works
+     * for btrfs as root filesystem.
+     */
+    class CommitCallbacks
+    {
+    public:
+
+	virtual ~CommitCallbacks() {}
+
+	virtual void post_root_filesystem_create() const {}
+	virtual void post_root_mount() const {}
+	virtual void post_root_fstab_add() const {}
```

Comment:
so it in fact mix empty base class and abstract class. Maybe it is related to discussion below. if you have something like

```
class CommitCallbacks
{
public:
  virtual ~CommitCallbacks() {}

  virtual void post_root_filesystem_create() const = 0;
}

// Null Callback
class EmptyCommitCallbacks
{
public:
  virtual ~CommitCallbacks() {}

  virtual void post_root_filesystem_create() const {}
  ...
}

EMPTY_COMMIT_CALLBACKS = EmptyCommitCallbacks.new();

class UserSpecifiedCallback : public EmptyCommitCallbacks
{
  // partial override
}
```
so this way user use `EMPTY_COMMIT_CALLBACKS` instead of `NULL` which is also slightly more readable when reading such code.